File: binary_search/kth_num.py
# https://www.acmicpc.net/problem/1300

N, k = int(input()), int(input())
start, end = 1, k
answer = 0

while start <= end:
    mid = (start + end) // 2

    temp = 0
    for i in range(1, N + 1):
        temp += min(mid // i, N)
    
    if temp >= k:
        answer = mid
        end = mid -1
    else:
        start = mid + 1

print(answer)

File: sort/kth_num.js
// https://www.acmicpc.net/problem/11004

const stdin = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : __dirname + "/../input.txt")
  .toString()
  .trim()
  .split("\n");
const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const [N, K] = input().split(" ").map(Number);
const Array = input()
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b);

console.log(Array[K - 1]);
